tests/testlib/simple_mock.nim

```diff
@@ -9,9 +9,9 @@ type Instr {.union.} = object
 proc mockImpl*(target, replacement: pointer) =
   # YOLO who needs alignment
   #doAssert (cast[ByteAddress](target) and ByteAddress(0x07)) == 0
-  var page = cast[pointer](cast[ByteAddress](target) and (not 0xfff))
+  var page = cast[pointer](cast[uint](target) and (not 0xfff))
   doAssert mprotect(page, 4096, PROT_WRITE or PROT_EXEC) == 0
-  let rel = cast[ByteAddress](replacement) - cast[ByteAddress](target) - 5
+  let rel = cast[uint](replacement) - cast[uint](target) - 5
```

Comment:
Why this?

Reply:
Because the original triggers deprecated warning (which suggests this fix).

Reply:
There are lots of other deprecation warnings, but unlike this one the fix is not suggested by the compiler, so I didn't touch those.
That should be done in a separate PR (ideally not by me, but by the nwaku team - I hope there's at least one person who cares about that).